### PR TITLE
Restrict players move when colliding with obstacles

### PR DESCRIPTION
**Pull request by pefontana**
# Restrict players move when colliding with obstacles

* Add walls to the map
* Change Players initial position
* Impl Debug trait to Rust structs
* Add entity.revert_move_entity() method
* Refactor fn collides_with to recibe a reference: entities: &Vec<Entity>
* Revert player move when colliding with an obstacle

### CODE REVIEW

File: priv/static/assets/app.css
Comment (manucamejo) [inline priv/static/assets/app.css:1]: Is this change necessary?

File: native/physics/src/lib.rs [hunk 1]
```diff
@@ -67,11 +69,19 @@ fn move_player(game_state: GameState, player_id: u64, x: f64, y: f64) -> GameSta
 }
 
 #[rustler::nif()]
-fn move_entities(game_state: GameState) -> GameState {
+fn move_entities(game_state: GameState, obstacles: Vec<Entity>) -> GameState {
     let mut game_state: GameState = game_state;
-
     for entity in game_state.entities.values_mut() {
+        // Avoid calculation for entities with no speed
+        if entity.speed == 0.0 {
```
Comment (manucamejo) [inline native/physics/src/lib.rs:76]: Can you add a direction condition? direction.x or direction.y should be different to 0.

File: lib/game_backend/game_updater.ex [hunk 2]
```diff
@@ -26,11 +26,106 @@ defmodule GameBackend.GameUpdater do
   def init(%{players: players}) do
     game_id = self() |> :erlang.term_to_binary() |> Base58.encode()
 
+    left_wall_vertices = [
```
Comment (manucamejo) [inline lib/game_backend/game_updater.ex:29]: In the future, this should be a config

**Review by manucamejo — CHANGES_REQUESTED** [review]

File: priv/static/assets/app.css
Comment (pefontana) [reply priv/static/assets/app.css:1]: When I compile the project that dependency made an upgrade
Should I restore it?

File: native/physics/src/lib.rs (hunk 1 above)
Comment (pefontana) [reply native/physics/src/lib.rs:76]: Sure I can add the check
But is not redundant? If the speed is 0, the player won't move, whether it has direction or no

File: priv/static/assets/app.css
Comment (manucamejo) [reply priv/static/assets/app.css:1]: I think you are doing a downgrade v3.3.2 -> v3.2.7

File: native/physics/src/lib.rs (hunk 1 above)
Comment (manucamejo) [reply native/physics/src/lib.rs:76]: Yes, but the speed is something inherent to the entity, it is the speed that it is going to move if it is moving. Right now it is the direction that determines whether it moves or not.
